infrastructure/pg/migrate.go

```diff
@@ -118,3 +118,74 @@ func (m *Migrate) Reset() error {
 
 	return nil
 }
+
+// Get the migration source files from Github, init the migrate and then run it.
+func InitAndRunMigrateFromGithub(
```

Comment:
Too much param, the caller side should be able to construct migrationSourceURL itself

Reply:
How about this one?
```
func (projection *Account) OnInit() error {
	migrationSourceURL := pg.GithubMigrationSourceURL(
		appprojection.MIGRATION_GITHUB_TARGET,
		projection.config.GithubAPIUser,
		projection.config.GithubAPIToken,
		MIGRATION_DIRECOTRY,
		projection.config.MigrationRepoRef,
	)
	databaseURL := pg.MigrationDBConnString(
		projection.rdbConn.(*pg.PgxConn).ConnString(),
		MIGRATION_TABLE_NAME,
	)
	if err := pg.InitAndRunMigrate(migrationSourceURL, databaseURL); err != nil {
		projection.logger.Errorf("failed to init and run migration: %v", err)
		return err
	}

	return nil
}
```

Reply:
Good and change `InitAndRunMigrate ` to `migrate`/`migrateFromGitHub`

Reply:
Sure, np~